sdks/java/core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java

```diff
@@ -121,10 +120,7 @@
     public ExecutorService create(PipelineOptions options) {
       ThreadFactoryBuilder threadFactoryBuilder = new ThreadFactoryBuilder();
       threadFactoryBuilder.setThreadFactory(MoreExecutors.platformThreadFactory());
-      if (!AppEngineEnvironment.IS_APP_ENGINE) {
-        // AppEngine doesn't allow modification of threads to be daemon threads.
-        threadFactoryBuilder.setDaemon(true);
-      }
+      threadFactoryBuilder.setDaemon(true);
```

Comment:
Reading the code, I think this PR is a no-op w.r.t. this issue, assuming we're not in App Engine. Am I missing something?